src/flamegraph/attrs.rs

```diff
@@ -0,0 +1,254 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{self, BufRead, BufReader};
+use std::path::PathBuf;
+
+/// Provides a way to customize the attributes on the SVG elements for a frame.
+#[derive(PartialEq, Eq, Debug, Default)]
+pub struct FuncFrameAttrsMap(HashMap<String, FrameAttrs>);
+
+impl FuncFrameAttrsMap {
+    /// Parse a FuncFrameAttrsMap from a file.
+    /// Each line should be a function name followed by a tab,
+    /// then a sequence of tab separated name=value pairs.
+    pub fn from_file(path: &PathBuf) -> io::Result<FuncFrameAttrsMap> {
+        let file = BufReader::new(File::open(path)?);
+        FuncFrameAttrsMap::from_reader(file)
+    }
+
+    /// Parse a FuncFrameAttrsMap from a reader.
+    /// Each line should be a function name followed by a tab,
+    /// then a sequence of tab separated name=value pairs.
+    pub fn from_reader<R: BufRead>(mut reader: R) -> io::Result<FuncFrameAttrsMap> {
+        let mut funcattr_map = FuncFrameAttrsMap::default();
+        let mut line = String::new();
+        loop {
+            line.clear();
+
+            if reader.read_line(&mut line)? == 0 {
+                break;
+            }
+
+            let mut line = line.trim().splitn(2, '\t');
+            if let Some(func) = line.next() {
```

Comment:
To avoid the extreme indentation here, I think I would use:
```rust
let func = if let Some(f) = line.next() {
  f
} else {
  continue;
};
```
We could probably make that nicer with a `try .. catch` and `Option?` + `continue` on `None`, but that's a bigger transformation.